src/plugins/mer/meremulatordevice.cpp

```diff
@@ -391,30 +363,22 @@ SshConnectionParameters MerEmulatorDevice::sshParametersForUser(const SshConnect
     return m_sshParams;
 }
 
-QMap<QString, MerEmulatorDeviceModel> MerEmulatorDevice::availableDeviceModels() const
-{
-#if __cplusplus < 201103L
-    const_cast<MerEmulatorDevice *>(this)->updateAvailableDeviceModels();
-#endif
-    return m_availableDeviceModels;
-}
-
 QString MerEmulatorDevice::deviceModel() const
 {
     return m_deviceModel;
 }
 
 void MerEmulatorDevice::setDeviceModel(const QString &deviceModel)
 {
-    QTC_CHECK(availableDeviceModels().contains(deviceModel));
+    QTC_CHECK(MerSettings::deviceModels().contains(deviceModel));
     QTC_CHECK(!virtualMachine().isEmpty());
 
     // Intentionally do not check for equal value - better for synchronization
     // with VB settings
 
     m_deviceModel = deviceModel;
 
-    QVariantMap fullDeviceModelData = readFullDeviceModelData();
+    const QVariantMap fullDeviceModelData = MerSettings::deviceModels().value(m_deviceModel).toMap();
     updateDconfDb(fullDeviceModelData);
```

Comment:
updateDconfDb should accept the dconfDb string directly now, so the code here should be `updateDconfDb(MerSettings::deviceModels().value(m_deviceModel).dconfDb)`

EDIT: now I noticed this is done in the last commit, so the change should be just moved to this commit.

Reply:
Done.